### DISCUSSION TITLE
Is it possible to renew the certificate 60 days before instead of 30 days?

### DISCUSSION THREAD

**Discussion by ankit-kumar-mck** [polymind-inc/acmebot#452] [General]
How to renew the certificate 60 days before instead of 30 days?
Using Let's encrypt to get certificate

**Comment by shibayan** [comment]
I do not currently implement customization of the renew timing, as certbot recommends 30 days before the update.

**Reply by ankit-kumar-mck** [reply]
Okay thanks for the update.
Is there any way to make changes direct in the code to update this setting?

**Reply by shibayan** [reply]
The change can be made by changing the code directly and deploying it.

If you have a clear reason for wanting 60 days, we can consider changing it as an option.

**Reply by ankit-kumar-mck** [reply]
Reason to change it after 60 days before: We have third party monitoring tool to monitor the certificate and notify to us if certificate expired 30 days before. So i want to renew the certificate every month instead of 30 days before.

**Reply by shibayan** [reply]
Thank you. I'll add the settings.

**Reply by ankit-kumar-mck** [reply]
Cool.
How soon you added these changes?
Waiting for your update.

**Reply by shibayan** [reply]
Very soon

**Reply by shibayan** [reply]
Released `v3.7.6`

**Reply by ankit-kumar-mck** [reply]
Thanks.

I just need to redeploy the function again using the https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2Fshibayan%2Fkeyvault-acmebot%2Fmaster%2Fazuredeploy.json to make that works?

Is this fetch latest code?

**Reply by shibayan** [reply]
No redeploy. Just restart Azure Functions and the latest version will be fetched.

**Reply by ankit-kumar-mck** [reply]
Thanks a lot

So after the restart Azure Functions no action required from my end. It will renew certificate 60 days before the expiry?

**Reply by shibayan** [reply]
Please see the following document to add the settings.
https://github.com/shibayan/keyvault-acmebot/wiki/Getting-Started#2-add-application-settings

**Reply by ankit-kumar-mck** [reply]
I have added configuration settings  --- > Acmebot:RenewBeforeExpiry with its value to 60.
But one of certificate expired in 51 days not renew automatically.

**Reply by ankit-kumar-mck** [reply]
@shibayan ^

**Reply by ankit-kumar-mck** [reply]
It's working.
But entry _Acmebot:RenewBeforeExpiry_ not visible in the configuration settings. It it removed automatically configuration settings?
